openedx/features/learner_analytics/views.py

```diff
@@ -144,25 +139,31 @@ def get_grade_data(self, user, course_key, grade_cutoffs):
         total_possible = 0
         # answered_percent seems to be unused and it does not take into account assignment type weightings
         answered_percent = None
-        for (location, subsection_grade) in course_grade.subsection_grades.iteritems():
```

Comment:
@nasthagiri: Not sure who to tag, but we _think_ that `course_grade.subsection_grades` is not ordered properly, where `course_grade.chapter_grades` seems to be, and getting the sections (subsections) from the chapter produces the correct order.

Reply:
Not sure why that would be since the implementation of `course_grade.subsection_grades` is exactly what you say your workaround is: https://github.com/edx/edx-platform/blob/a600ef4c25f8967ed95848410b0b512d0b587b2b/lms/djangoapps/grades/course_grade.py#L90 

The only difference may be that the `course_grade.subsection_grades` method has a `lazy` attribute.

Reply:
I didn't look at the code and @dianakhuang did say it was confusing why this was happening.  However, we have a bug we are fixing, so something is going wrong. :)